packages/discord.js/src/structures/GuildInvite.js

```diff
@@ -0,0 +1,183 @@
+'use strict';
+
+const { Routes, PermissionFlagsBits, InviteType } = require('discord-api-types/v10');
+const { BaseInvite } = require('./BaseInvite.js');
+const { GuildScheduledEvent } = require('./GuildScheduledEvent.js');
+const { IntegrationApplication } = require('./IntegrationApplication.js');
+const { InviteGuild } = require('./InviteGuild.js');
+const { DiscordjsError, ErrorCodes } = require('../errors/index.js');
+
+/**
+ * A channel invite leading to a guild.
+ * @extends {BaseInvite}
+ */
+class GuildInvite extends BaseInvite {
+  constructor(client, data) {
+    super(client, data);
+
+    // Type may be missing from audit logs.
+    this.type ??= InviteType.Guild;
+
+    /**
+     * The id of the guild this invite is for.
+     * @type {Snowflake}
+     */
+    // Guild id may be missing from audit logs.
+    this.guildId = data.guild_id ?? data.guild.id;
+
+    /**
+     * The maximum age of the invite in seconds. `0` for no expiry.
+     * <info>This is only available when the invite was fetched through {@link GuildInviteManager#fetch}
+     * or created through {@link GuildInviteManager#create}.</info>
+     * @name GuildInvite#maxAge
+     * @type {?number}
+     */
+
+    /**
+     * The approximate total number of members of the guild.
+     * <info>This is only available when the invite was fetched through {@link Client#fetchInvite}.</info>
+     * @name GuildInvite#approximateMemberCount
+     * @type {?number}
+     */
+  }
+
+  _patch(data) {
+    if ('guild' in data) {
+      /**
+       * The guild the invite is for. May include welcome screen data.
+       * @type {?(Guild|InviteGuild)}
+       */
+      this.guild = this.client.guilds.cache.get(data.guild.id) ?? new InviteGuild(this.client, data.guild);
+    } else {
+      this.guild ??= null;
+    }
+
+    super._patch(data, this.guild);
```

Comment:
This is a confusing approach. Super constructor calls this._patch which actually uses the implementation here, which then (after assigning the guild from data) call super._patch(), passing that guild. BaseInvite should have no business with that guild, since only GuildInvites have them.